packages/hop-node/src/cli/stake.ts

```diff
@@ -1,66 +1,137 @@
-import StakeWatcher from 'src/watchers/StakeWatcher'
+import chainSlugToId from 'src/utils/chainSlugToId'
+import BondWithdrawalWatcher from 'src/watchers/BondWithdrawalWatcher'
+import L1Bridge from 'src/watchers/classes/L1Bridge'
+import L2Bridge from 'src/watchers/classes/L2Bridge'
+import Token from 'src/watchers/classes/Token'
+import { BigNumber, constants } from 'ethers'
 import { Chain } from 'src/constants'
 import {
   findWatcher,
-  getStakeWatchers
+  getWatchers
 } from 'src/watchers/watchers'
 import {
   parseConfigFile,
   setGlobalConfigFromConfigFile
 } from 'src/config'
-
+import wait from 'src/utils/wait'
 import { logger, program } from './shared'
 
-export enum StakerAction {
-  Stake,
-  Unstake,
-  Status
+async function sendTokensToL2 (
+  bridge: L1Bridge,
+  parsedAmount: BigNumber,
+  chain: string
+) {
+  const spender = bridge.getAddress()
+  const token: Token | void = await getToken(bridge)
+
+  let tx
+  if (token) {
+    logger.debug('Approving L2 token send, if needed')
+    tx = await token.approve(spender, parsedAmount)
+    await tx?.wait()
+  }
+
+  logger.debug('Sending tokens to L2')
+  tx = await bridge.convertCanonicalTokenToHopToken(
+    chainSlugToId(chain)!,
+    parsedAmount
+  )
+  await tx.wait()
 }
 
-export async function staker (
-  chain: string,
-  token: string,
-  amount: number,
-  action: StakerAction
+async function stake (
+  bridge: L2Bridge | L1Bridge,
+  parsedAmount: BigNumber
 ) {
-  if (!chain) {
+  logger.debug('Staking')
+
+  const token: Token | void = await getToken(bridge)
+  const stakeTokenBalance: BigNumber = await getTokenBalance(bridge, token)
+  const formattedAmount = bridge.formatUnits(parsedAmount)
```

Comment:
do decimals need to be passed here for `formatUnits`?

Reply:
It is my understanding that the `bridge` [does this already](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/watchers/classes/Bridge.ts#L506-L512) with the `token` associated with the bridge itself.

Reply:
you're right. Disregard comment